lib/chutney/TorNet.py

```diff
@@ -1355,8 +1364,9 @@ def getNodeDirInfoStatusPattern(self, dir_format):
         elif desc:
             return r'^router ' + nickname + " "
         elif md:
+            
             # Not yet implemented, see #33428
-            return r'^id ed25519 ' + re.escape('ed25519_id')
+            return r'^id ed25519 ' + re.escape(ed25519_key)
```

Comment:
I was getting error for nonetype in the code earlier, so I assigned a variable ed25519_key some random value , the error did not appear again, I was thinking maybe i have to handle the none type error by the following code

 if ed25519_key = None:

     return r'^id ed25519 ' + ed25519_key
else:

    return  r'^id ed25519 ' + re.escape(ed25519_key)"

Reply:
Do you think I am right with the above code?

Reply:
> I was getting error for nonetype in the code earlier, so I assigned a variable ed25519_key some random value , the error did not appear again

Which code are you running to get the logs you sent me?
I need to see the exact code that you are running.

If I don't have the same code, I can't run it myself.
So it is much harder to help you.

> I was thinking maybe i have to handle the none type error by the following code
> 
> if ed25519_key = None:
> 
> ```
>  return r'^id ed25519 ' + ed25519_key
> ```
> 
> else:
> 
> ```
> return  r'^id ed25519 ' + re.escape(ed25519_key)"
> ```

Can you explain what you think this code does?

What value should you return if ed25519_key is None?
What does the calling code expect?
What do the other parts of this function return, when they can't make a pattern?

Reply:
The log is based on the code of recent commit.